From 565ceb4a6bc8e7839905c45ed544af16ab18f9e6 Mon Sep 17 00:00:00 2001
From: Naveen Gangam <ngangam@apache.org>
Date: Tue, 9 May 2017 06:04:09 -0400
Subject: [PATCH 1124/1164] CDH-53988 : HIVE-16556 : Hive Metastore schema
 changes to add METASTORE_DB_PROPERTIES table
 (Vihang Karajgaonkar, reviewed by Naveen Gangam)

Change-Id: I0ec1009d34d3586ff1572f04ceb9993d527ddeb3
---
 .../scripts/upgrade/derby/041-HIVE-16556.derby.sql |    5 +++++
 .../scripts/upgrade/mssql/026-HIVE-16556.mssql.sql |   10 ++++++++++
 .../scripts/upgrade/mysql/041-HIVE-16556.mysql.sql |   11 +++++++++++
 .../upgrade/oracle/041-HIVE-16556.oracle.sql       |   11 +++++++++++
 .../upgrade/postgres/040-HIVE-16556.postgres.sql   |   13 +++++++++++++
 5 files changed, 50 insertions(+)
 create mode 100644 metastore/scripts/upgrade/derby/041-HIVE-16556.derby.sql
 create mode 100644 metastore/scripts/upgrade/mssql/026-HIVE-16556.mssql.sql
 create mode 100644 metastore/scripts/upgrade/mysql/041-HIVE-16556.mysql.sql
 create mode 100644 metastore/scripts/upgrade/oracle/041-HIVE-16556.oracle.sql
 create mode 100644 metastore/scripts/upgrade/postgres/040-HIVE-16556.postgres.sql

diff --git a/metastore/scripts/upgrade/derby/041-HIVE-16556.derby.sql b/metastore/scripts/upgrade/derby/041-HIVE-16556.derby.sql
new file mode 100644
index 0000000..ea5f24a
--- /dev/null
+++ b/metastore/scripts/upgrade/derby/041-HIVE-16556.derby.sql
@@ -0,0 +1,5 @@
+CREATE TABLE "APP"."METASTORE_DB_PROPERTIES" ("PROPERTY_ID" BIGINT NOT NULL, "PROPERTY_KEY" VARCHAR(255) NOT NULL, "PROPERTY_VALUE" VARCHAR(1000) NOT NULL, "DESCRIPTION" VARCHAR(1000));
+
+ALTER TABLE "APP"."METASTORE_DB_PROPERTIES" ADD CONSTRAINT "UNIQUE_PROPERTY_KEY" UNIQUE ("PROPERTY_KEY");
+
+ALTER TABLE "APP"."METASTORE_DB_PROPERTIES" ADD CONSTRAINT "PROPERTY_ID_PK" PRIMARY KEY ("PROPERTY_ID");
diff --git a/metastore/scripts/upgrade/mssql/026-HIVE-16556.mssql.sql b/metastore/scripts/upgrade/mssql/026-HIVE-16556.mssql.sql
new file mode 100644
index 0000000..088e702
--- /dev/null
+++ b/metastore/scripts/upgrade/mssql/026-HIVE-16556.mssql.sql
@@ -0,0 +1,10 @@
+CREATE TABLE METASTORE_DB_PROPERTIES (
+  PROPERTY_ID BIGINT NOT NULL,
+  PROPERTY_KEY VARCHAR(255) NOT NULL,
+  PROPERTY_VALUE VARCHAR(1000) NOT NULL,
+  DESCRIPTION VARCHAR(1000)
+);
+
+ALTER TABLE METASTORE_DB_PROPERTIES ADD CONSTRAINT METASTORE_DB_PROPERTIES_PK PRIMARY KEY (PROPERTY_ID);
+
+ALTER TABLE METASTORE_DB_PROPERTIES ADD CONSTRAINT UNIQUE_PROPERTY_KEY UNIQUE (PROPERTY_KEY);
diff --git a/metastore/scripts/upgrade/mysql/041-HIVE-16556.mysql.sql b/metastore/scripts/upgrade/mysql/041-HIVE-16556.mysql.sql
new file mode 100644
index 0000000..8afc196
--- /dev/null
+++ b/metastore/scripts/upgrade/mysql/041-HIVE-16556.mysql.sql
@@ -0,0 +1,11 @@
+--
+-- Table structure for table METASTORE_DB_PROPERTIES
+--
+CREATE TABLE IF NOT EXISTS `METASTORE_DB_PROPERTIES` (
+  `PROPERTY_ID` BIGINT(20) NOT NULL,
+  `PROPERTY_KEY` varchar(255) NOT NULL,
+  `PROPERTY_VALUE` varchar(1000) NOT NULL,
+  `DESCRIPTION` varchar(1000),
+ PRIMARY KEY(`PROPERTY_ID`),
+ UNIQUE KEY `UNIQUE_PROPERTY_KEY` (`PROPERTY_KEY`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
diff --git a/metastore/scripts/upgrade/oracle/041-HIVE-16556.oracle.sql b/metastore/scripts/upgrade/oracle/041-HIVE-16556.oracle.sql
new file mode 100644
index 0000000..14a2bdb
--- /dev/null
+++ b/metastore/scripts/upgrade/oracle/041-HIVE-16556.oracle.sql
@@ -0,0 +1,11 @@
+CREATE TABLE METASTORE_DB_PROPERTIES
+(
+  PROPERTY_ID NUMBER NOT NULL,
+  PROPERTY_KEY VARCHAR(255) NOT NULL,
+  PROPERTY_VALUE VARCHAR(1000) NOT NULL,
+  DESCRIPTION VARCHAR(1000)
+);
+
+ALTER TABLE METASTORE_DB_PROPERTIES ADD CONSTRAINT UNIQUE_PROPERTY_KEY UNIQUE (PROPERTY_KEY);
+
+ALTER TABLE METASTORE_DB_PROPERTIES ADD CONSTRAINT PROPERTY_ID_PK PRIMARY KEY (PROPERTY_ID);
diff --git a/metastore/scripts/upgrade/postgres/040-HIVE-16556.postgres.sql b/metastore/scripts/upgrade/postgres/040-HIVE-16556.postgres.sql
new file mode 100644
index 0000000..9a3ff03
--- /dev/null
+++ b/metastore/scripts/upgrade/postgres/040-HIVE-16556.postgres.sql
@@ -0,0 +1,13 @@
+CREATE TABLE "METASTORE_DB_PROPERTIES"
+(
+  "PROPERTY_ID" BIGINT NOT NULL,
+  "PROPERTY_KEY" VARCHAR(255) NOT NULL,
+  "PROPERTY_VALUE" VARCHAR(1000) NOT NULL,
+  "DESCRIPTION" VARCHAR(1000)
+);
+
+ALTER TABLE ONLY "METASTORE_DB_PROPERTIES"
+  ADD CONSTRAINT "UNIQUE_PROPERTY_KEY" UNIQUE ("PROPERTY_KEY");
+
+ALTER TABLE ONLY "METASTORE_DB_PROPERTIES"
+  ADD CONSTRAINT "METASTORE_DB_PROPERTIES_pkey" PRIMARY KEY ("PROPERTY_ID");
-- 
1.7.9.5

